python/tests/test_conversions.py

```diff
@@ -0,0 +1,442 @@
+        - 345 is 0...1|01011001 in binary
+        - 00000001 -> 1, 01011001 -> 89
+    Float:
+        - Stored as 4 bytes in little-endian order
+        - floating point numbers are represented as sign * 2ˆexponent * mantissa
+        - -4.5F is -1 * 2ˆ2 * 1.125 and encoded as 11000000|10010000|0...0 in binary
+        - 00000000 -> 0, 00000000 -> 0, 10010000 -> 144 (-112), 11000000 -> 192 (-64),
+"""
+import struct
+import uuid
+from decimal import Decimal
+
+import pytest
+
+from iceberg import conversions
+from iceberg.types import (
+    BinaryType,
+    BooleanType,
+    DateType,
+    DecimalType,
+    DoubleType,
+    FixedType,
+    FloatType,
+    IntegerType,
+    LongType,
+    StringType,
+    TimestampType,
+    TimestamptzType,
+    TimeType,
+    UUIDType,
+)
+
+
+@pytest.mark.parametrize(
+    "primitive_type, partition_value_as_str, expected_result",
+    [
+        (BooleanType(), "true", True),
+        (BooleanType(), "false", False),
+        (BooleanType(), "TRUE", True),
+        (BooleanType(), "FALSE", False),
+        (IntegerType(), "1", 1),
+        (IntegerType(), "9999", 9999),
+        (LongType(), "123456789", 123456789),
+        (FloatType(), "1.1", 1.1),
+        (DoubleType(), "99999.9", 99999.9),
+        (DecimalType(5, 2), "123.45", Decimal("123.45")),
+        (StringType(), "foo", "foo"),
+        (UUIDType(), "f79c3e09-677c-4bbd-a479-3f349cb785e7", uuid.UUID("f79c3e09-677c-4bbd-a479-3f349cb785e7")),
+        (FixedType(3), "foo", b"foo"),
+        (BinaryType(), "foo", b"foo"),
+        (None, None, None),
+    ],
+)
+def test_from_partition_value_to_py(primitive_type, partition_value_as_str, expected_result):
+    """Test converting a partition value to a python built-in"""
+    assert conversions.from_partition_value_to_py(primitive_type, partition_value_as_str) == expected_result
+
+
+@pytest.mark.parametrize(
+    "primitive_type",
+    [
+        (BinaryType()),
+        (BooleanType()),
+        (DateType()),
+        (DecimalType(2, 1)),
+        (DoubleType()),
+        (FixedType(1)),
+        (FloatType()),
+        (IntegerType()),
+        (LongType()),
+        (StringType()),
+        (TimestampType()),
+        (TimestamptzType()),
+        (TimeType()),
+        (UUIDType()),
+    ],
+)
+def test_none_partition_values(primitive_type):
+    """Test converting a partition value to a python built-in"""
+    assert conversions.from_partition_value_to_py(primitive_type, None) == None
+
+
+@pytest.mark.parametrize(
+    "primitive_type, b, result",
+    [
+        (BooleanType(), b"\x00", False),
+        (BooleanType(), b"\x01", True),
+        (IntegerType(), b"\xd2\x04\x00\x00", 1234),
+        (LongType(), b"\xd2\x04\x00\x00\x00\x00\x00\x00", 1234),
+        (DoubleType(), b"\x8d\x97\x6e\x12\x83\xc0\xf3\x3f", 1.2345),
+        (DateType(), bytes([232, 3, 0, 0]), 1000),
+        (DateType(), b"\xd2\x04\x00\x00", 1234),
```

Comment:
What is the purpose of testing both `bytes` representations? Is this to match existing tests in other places?

Reply:
Replaced these with byte strings. I don't think there's any purpose for having both representations or even the tests that validate that the resulting strings are utf-8, these were brought over from the legacy library. I think they may have been relevant at the time because python 2 was still being supported which had some different encoding behaviors compared to python 3+.